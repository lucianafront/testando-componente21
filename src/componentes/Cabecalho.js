import Botao from "./Botao";

function Cabecalho() {
    return (

        <div className="container">
            <div className="nav-container">
                <div className="brand">
                    <a href="https://sodiedoces.com.br/">

                        <img id="img-logo" src="https://sodiedoces.com.br/wp-content/themes/sodiedoces/img/logo-sodie.png" alt="Sodiê Doces" /></a>
                </div>
                <nav>
                    <div className="nav-mobile"><a id="nav-toggle"><span></span></a></div>
                    <ul className="nav-list">
                        <li><a href="https://sodiedoces.com.br/sobre">A Sodiê</a></li>
                        <li><a id="abrirsub" href="#!">Produtos <i className="fas fa-caret-down" aria-hidden="true"></i> </a>
                            <ul className="nav-dropdown">
                                <li><a href="https://sodiedoces.com.br/bolos">Bolos</a></li>
                                <li><a href="https://sodiedoces.com.br/salgados">Salgados</a></li>
                                <li><a href="https://sodiedoces.com.br/kit-festa/escolha-seu-kit-festa">Kit Festa</a></li>
                                <li><a href="https://sodiedoces.com.br/doces">Doces</a></li>
                                <li><a href="https://sodiedoces.com.br/tortas-doces">Tortas Doces</a></li>
                                <li><a href="https://sodiedoces.com.br/bebidas">Bebidas</a></li>
                                <li><a href="https://sodiedoces.com.br/balas-de-coco">Balas de Coco</a></li>
                            </ul>
                        </li>
                        <li><a href="https://sodiedoces.com.br/lojas">Lojas</a></li>
                        <li><a href="https://sodiedoces.com.br/contato">Contato</a></li>
                    </ul>
                </nav>
                <div className="brand2 center">

                    <div className="nav-app center btn-marrom">
                        <p>Peça no app</p>

                        <Botao
                            link='https://play.google.com/store/apps/details?id=com.sodiedoces'
                            ico='fab fa-google-play'
                        />
                        
                         <Botao
                         link='https://apps.apple.com/br/app/sodi%C3%AA-doces-oficial/id1557040152'
                         ico='fab fa-apple'

                         />
                        
                    </div>
                    <Botao
                    link='https://suafranquia.sodiedoces.com.br/oportunid'
                    ico='nav-franqueado center btn-marrom'
                    />


                    <ul className="nav-social ulflex">
                        <li>
                            <Botao
                            link='ttps://instagram.com/SodieDo'
                            ico='fab fa-instagram'
                            />
                            <Botao
                            link='https://instagram.com/SodieDoces'
                            ico='fab fa-instagram'
                            />
                            <Botao
                            link='https://instagram.com/SodieDoces'
                            ico='fab fa-instagram'
                            />

                              
                        </li>
                        <li>

                            <Botao
                             link='https://www.facebook.com/sodiedocesofici'
                             ico='ab fa-facebook-square'
                             />
                            
                        </li>
                    </ul>
                </div>
            </div>
        </div>

    );
}

export default Cabecalho;
