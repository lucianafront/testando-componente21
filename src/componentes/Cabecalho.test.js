import React from 'react';
import { render } from '@testing-library/react';
import Cabecalho from './componentes/Cabecalho';

test('renders Cabecalho component correctly', () => {
  const { asFragment } = render(<Cabecalho />);
  expect(asFragment()).toMatchSnapshot();
});

test('calls the correct functions on link clicks', () => {
    // Mocking the functions
    window.open = jest.fn();
  
    const { getByText } = render(<Cabecalho />);
  
    // Simulating link clicks
    fireEvent.click(getByText('A Sodiê'));
    fireEvent.click(getByText('Produtos'));
    fireEvent.click(getByText('Bolos'));
    fireEvent.click(getByText('Salgados'));
    fireEvent.click(getByText('Kit Festa'));
    fireEvent.click(getByText('Doces'));
    fireEvent.click(getByText('Tortas Doces'));
    fireEvent.click(getByText('Bebidas'));
    fireEvent.click(getByText('Balas de Coco'));
    fireEvent.click(getByText('Lojas'));
    fireEvent.click(getByText('Contato'));
    fireEvent.click(getByText('Peça no app'));
    fireEvent.click(getByText('Seja um franqueado'));
  
    // Checking if the functions were called
    expect(window.open).toHaveBeenCalledTimes(13);
  });
