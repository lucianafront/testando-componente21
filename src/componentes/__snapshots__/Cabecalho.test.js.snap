// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Cabecalho component correctly 1`] = `
<DocumentFragment>
  <div
    class="container"
  >
    <div
      class="nav-container"
    >
      <div
        class="brand"
      >
        <a
          href="https://sodiedoces.com.br/"
        >
          <img
            alt="Sodiê Doces"
            id="img-logo"
            src="https://sodiedoces.com.br/wp-content/themes/sodiedoces/img/logo-sodie.png"
          />
        </a>
      </div>
      <nav>
        <div
          class="nav-mobile"
        >
          <a
            id="nav-toggle"
          >
            <span />
          </a>
        </div>
        <ul
          class="nav-list"
        >
          <li>
            <a
              href="https://sodiedoces.com.br/sobre"
            >
              A Sodiê
            </a>
          </li>
          <li>
            <a
              href="#!"
              id="abrirsub"
            >
              Produtos 
              <i
                aria-hidden="true"
                class="fas fa-caret-down"
              />
               
            </a>
            <ul
              class="nav-dropdown"
            >
              <li>
                <a
                  href="https://sodiedoces.com.br/bolos"
                >
                  Bolos
                </a>
              </li>
              <li>
                <a
                  href="https://sodiedoces.com.br/salgados"
                >
                  Salgados
                </a>
              </li>
              <li>
                <a
                  href="https://sodiedoces.com.br/kit-festa/escolha-seu-kit-festa"
                >
                  Kit Festa
                </a>
              </li>
              <li>
                <a
                  href="https://sodiedoces.com.br/doces"
                >
                  Doces
                </a>
              </li>
              <li>
                <a
                  href="https://sodiedoces.com.br/tortas-doces"
                >
                  Tortas Doces
                </a>
              </li>
              <li>
                <a
                  href="https://sodiedoces.com.br/bebidas"
                >
                  Bebidas
                </a>
              </li>
              <li>
                <a
                  href="https://sodiedoces.com.br/balas-de-coco"
                >
                  Balas de Coco
                </a>
              </li>
            </ul>
          </li>
          <li>
            <a
              href="https://sodiedoces.com.br/lojas"
            >
              Lojas
            </a>
          </li>
          <li>
            <a
              href="https://sodiedoces.com.br/contato"
            >
              Contato
            </a>
          </li>
        </ul>
      </nav>
      <div
        class="brand2 center"
      >
        <div
          class="nav-app center btn-marrom"
        >
          <p>
            Peça no app
          </p>
          <a
            href="https://play.google.com/store/apps/details?id=com.sodiedoces"
            rel="noreferrer"
            target="_blank"
          >
            <i
              aria-hidden="true"
              class="fab fa-google-play"
            />
          </a>
          <a
            href="https://apps.apple.com/br/app/sodi%C3%AA-doces-oficial/id1557040152"
            rel="noreferrer"
            target="_blank"
          >
            <i
              aria-hidden="true"
              class="fab fa-apple"
            />
          </a>
        </div>
        <a
          href="https://suafranquia.sodiedoces.com.br/oportunid"
          rel="noreferrer"
          target="_blank"
        >
          <i
            aria-hidden="true"
            class="nav-franqueado center btn-marrom"
          />
        </a>
        <ul
          class="nav-social ulflex"
        >
          <li>
            <a
              href="ttps://instagram.com/SodieDo"
              rel="noreferrer"
              target="_blank"
            >
              <i
                aria-hidden="true"
                class="fab fa-instagram"
              />
            </a>
            <a
              href="https://instagram.com/SodieDoces"
              rel="noreferrer"
              target="_blank"
            >
              <i
                aria-hidden="true"
                class="fab fa-instagram"
              />
            </a>
            <a
              href="https://instagram.com/SodieDoces"
              rel="noreferrer"
              target="_blank"
            >
              <i
                aria-hidden="true"
                class="fab fa-instagram"
              />
            </a>
          </li>
          <li>
            <a
              href="https://www.facebook.com/sodiedocesofici"
              rel="noreferrer"
              target="_blank"
            >
              <i
                aria-hidden="true"
                class="ab fa-facebook-square"
              />
            </a>
          </li>
        </ul>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
